tests/nlu/tokenizers/test_jieba_tokenizer.py

```diff
@@ -36,18 +50,62 @@ def test_jieba(text, expected_tokens, expected_indices):
     assert [t.end for t in tokens] == [i[1] for i in expected_indices]
 
 
-def test_jieba_load_dictionary(tmp_path: Path):
-    dictionary_path = str(tmp_path)
-
-    component_config = {"dictionary_path": dictionary_path}
-
-    with patch.object(
-        JiebaTokenizer, "load_custom_dictionary", return_value=None
-    ) as mock_method:
-        tk = JiebaTokenizer(component_config)
-        tk.tokenize(Message(data={TEXT: ""}), attribute=TEXT)
-
-    mock_method.assert_called_once_with(dictionary_path)
+def test_jieba_load_and_persist_dictionary(
+    tmp_path_factory: TempPathFactory,
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+):
+    dictionary_directory = tmp_path_factory.mktemp("dictionaries")
+    dictionary_path = dictionary_directory / "dictionary_1"
+
+    dictionary_contents = """
+创新办 3 i
+云计算 5
+凱特琳 nz
+台中
+        """
+    dictionary_path.write_text(dictionary_contents)
+
+    component_config = {"dictionary_path": dictionary_directory}
+
+    resource = Resource("jieba")
+    tk = JiebaTokenizerGraphComponent.create(
+        {**JiebaTokenizerGraphComponent.get_default_config(), **component_config},
+        default_model_storage,
+        resource,
+        default_execution_context,
+    )
+
+    tk.process_training_data(TrainingData([Message(data={TEXT: ""})]))
+
+    # The dictionary has not been persisted yet.
+    with pytest.raises(ValueError):
+        JiebaTokenizerGraphComponent.load(
+            {**JiebaTokenizerGraphComponent.get_default_config(), **component_config},
+            default_model_storage,
+            resource,
+            default_execution_context,
+        )
+
+    tk.persist()
```

Comment:
shouldn't this be called as part of `train`?

Reply:
This is what I was asking about in the slack thread. There isn't really a `train`, only a `process_training_data`. We could save it as part of that method, but it makes it a bit different from the usual patten. I don't really feel strongly either way though.

Reply:
same. Probably having separate things is cleaner tbh.

Reply:
ok I will leave it separate for now, but it's a quick change later if we find it's easier w.r.t to the recipe (or whatever) to have them combined.

Reply:
Can we do a train and load test then?

Reply:
I'm not sure what you mean. This test does, `process_training_data`, `persist`, `load` then `process` which is the full lifecycle.

Reply:
sorry, all good 🙌🏻